osu.Game/Configuration/SettingSourceAttribute.cs

```diff
@@ -139,9 +139,12 @@ public static IEnumerable<(SettingSourceAttribute, PropertyInfo)> GetSettingsSou
 
         public static IEnumerable<(SettingSourceAttribute, PropertyInfo)> GetOrderedSettingsSourceProperties(this object obj)
         {
-            var original = obj.GetSettingsSourceProperties();
+            var original = obj.GetSettingsSourceProperties().ToArray();
 
-            var orderedRelative = original.Where(attr => attr.Item1.OrderPosition != null).OrderBy(attr => attr.Item1.OrderPosition);
+            var orderedRelative = original
+                                  .Where(attr => attr.Item1.OrderPosition != null)
+                                  .OrderBy(attr => attr.Item1.OrderPosition)
+                                  .ToArray();
             var unordered = original.Except(orderedRelative);
 
             return orderedRelative.Concat(unordered);
```

Comment:
I'm not sure about this. Shouldn't we also `ToArray` the return here? Right now it's passing a two-level linq operation, which feels pretty prone to (at very least) performance issues.

Reply:
Or maybe just replace this (and the whole `unordered` weirdness) with non-linq code (changing `original` to `ToList` and using `Contains` or similar).

Reply:
I can do that (and change to `ICollection`). I think there's also a way to write this with two `OrderBy`s and avoid allocating one array for `orderedRelative`, but the way I have in mind isn't super readable (it'd be using `OrderByDescending` on a bool which feels weird).

Reply:
After some thinking I've settled on making `SettingSourceAttribute` implement `IComparable`, which carries the "first by order value, then unordered" logic. Let me know how you're feeling about that.